src/concrete/ml/quantization/quantized_ops.py

```diff
@@ -491,6 +491,68 @@ def copy_function(x):
             params=self.output_quant_params,
         )
 
+    def calibrate_analytical_output(self, *inputs: QuantizedArray) -> UniformQuantizer:
+        """Calibrate output quantization based on analytical formulas.
+
+        Args:
+            *inputs (QuantizedArray): quantized operation inputs. Quantized weights
+                are stored in the op instance
+
+        Returns:
+            res (UniformQuantizer): the quantizer of the operation's output values
+                that can be used to de-quantize these values.
+        """
+
+        q_input1 = inputs[0]
+        assert isinstance(q_input1, QuantizedArray)
+        q_input2 = inputs[1]
+        assert isinstance(q_input2, QuantizedArray)
+
+        # In the operation Y = alpha * A' * B' + beta * C, q_bias is used for
+        # generalised matrix multiplication. q_bias is set to None for standard
+        # matrix multiplication (beta == 0 or only two inputs)
+        q_bias = None if len(inputs) == 2 or self.attrs["beta"] == 0 else inputs[2]
+        assert isinstance(q_bias, (type(None), QuantizedArray))
+
+        # Using snake case here to please the Python format, the original attrs don't have the '_'
+        # Use default false so we also support MatMul impl, MatMul does not have these flags
+        transpose_inputs2 = self.attrs.get("transB", False)
+
+        p = q_input2.qvalues.shape[-2]
+
+        assert q_input1.quantizer.scale is not None
+        assert q_input2.quantizer.scale is not None
+        m_matmul = q_input1.quantizer.scale * q_input2.quantizer.scale
+
```

Comment:
I would have added a comment to show the formula of out_zp (even though it's trivial, but it might help in the future):

out_zp = sum_weights − final_term
out_zp = zp_input1 ​× ∑input2_i − p × zp_input1 ​× zp_input2